android/src/main/java/com/klippa/reactscanner/KlippaScannerSDKModule.kt

```diff
@@ -310,8 +305,81 @@ class KlippaScannerSDKModule(
 
-            val klippaScanner = builder.build(reactContext)
-            currentActivity.startActivity(klippaScanner)
+            val modes = ArrayList<KlippaDocumentMode>()
+
+            if (config.hasKey("CameraModeSingle")) {
+                val cameraModeSingle = config.getMap("CameraModeSingle") ?: return
+
+                var name = "Single Document"
+                if (cameraModeSingle.hasKey("name")) {
+                    name = cameraModeSingle.getString("name").toString()
+                }
+
+                var message = "Single Document"
+                if (cameraModeSingle.hasKey("message")) {
+                    message = cameraModeSingle.getString("message").toString()
+                }
+
+                val singleCameraMode = KlippaSingleDocumentMode(name = name, Instructions(message = message, dismissHandler = {
+                    singleDocumentModeInstructionsDismissed = true
+                }))
+
+                modes.add(singleCameraMode)
+            }
+
+            if (config.hasKey("CameraModeMulti")) {
+                val cameraMode = config.getMap("CameraModeMulti") ?: return
+
+                var name = "Multiple Document Mode"
+                if (cameraMode.hasKey("name")) {
+                    name = cameraMode.getString("name").toString()
+                }
+
+                var message = "Multiple Document Mode"
+                if (cameraMode.hasKey("message")) {
+                    message = cameraMode.getString("message").toString()
+                }
+
+                val multipleCameraMode = KlippaMultipleDocumentMode(name = name, Instructions(message = message, dismissHandler = {
+                    multiDocumentModeInstructionsDismissed = true
+                }))
+
+                modes.add(multipleCameraMode)
+            }
+
+            if (config.hasKey("CameraModeSegmented")) {
+                val cameraMode = config.getMap("CameraModeSegmented") ?: return
+
+                var name = "Segmented Document"
+                if (cameraMode.hasKey("name")) {
+                    name = cameraMode.getString("name").toString()
+                }
+
+                var message = "Segmented Document"
+                if (cameraMode.hasKey("message")) {
+                    message = cameraMode.getString("message").toString()
+                }
+
+                val segmentedCameraMode = KlippaSegmentedDocumentMode(name = name, Instructions(message = message, dismissHandler = {
+                    segmentedDocumentModeInstructionsDismissed = true
+                }))
+
+                modes.add(segmentedCameraMode)
+            }
+
+            var index = 0
+            if (config.hasKey("StartingIndex")) {
+                index = config.getInt("StartingIndex")
+            }
+
+            val cameraModes = KlippaCameraModes(
+                    modes = modes,
+                    startingIndex = index
+            )
+
+            builder.cameraModes = cameraModes
```

Comment:
if modes is empty don't set it on the builder, the builder will set default values if no modes are supplied.

Reply:
Added `modes.isEmpty` check to both iOS and Android versions